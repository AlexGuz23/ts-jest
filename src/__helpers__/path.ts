import { resolve, relative } from 'path'

export const ROOT = resolve(__dirname, '..', '..')

export function relativeToRoot(path: string) {
  return relative(ROOT, path)
}
