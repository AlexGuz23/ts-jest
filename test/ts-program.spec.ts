import TsProgram from '../src/ts-program';
import { resolve } from 'path';

const path = '/dummy/path/to/file.ts';
const content = `
import upper from './upper';
import lower from './lower';

jest.mock('./upper', () => (s) => s.toUpperCase());

describe('hello', () => {
  test('my test', () => {
    expect(upper('hello')).toBe('HELLO');
    expect(lower('HELLO')).toBe('hello');
    jest.mock('./lower', () => (s) => s.toLowerCase());
  });
});
`;

describe('hoisting', () => {
  const prog = new TsProgram(resolve(__dirname, '..'), {
    useBabelJest: false,
    inputOptions: {},
    diagnostics: [],
  });

  it('should hoist jest.mock()', () => {
    const result = prog.transpileModule(path, content, undefined, {
      inlineSourceMap: false,
    });
    expect(result).toMatchSnapshot();
  });
});
