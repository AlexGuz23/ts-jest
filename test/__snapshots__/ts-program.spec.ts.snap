// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hoisting should hoist jest.mock() 1`] = `
"\\"use strict\\";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { \\"default\\": mod };
};
Object.defineProperty(exports, \\"__esModule\\", { value: true });
jest.mock('./upper', function () { return function (s) { return s.toUpperCase(); }; });
jest.mock('./lower', function () { return function (s) { return s.toLowerCase(); }; });
var upper_1 = __importDefault(require(\\"./upper\\"));
var lower_1 = __importDefault(require(\\"./lower\\"));
describe('hello', function () {
    test('my test', function () {
        expect(upper_1.default('hello')).toBe('HELLO');
        expect(lower_1.default('HELLO')).toBe('hello');
    });
});
"
`;
